src/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs

```diff
@@ -45,6 +47,19 @@ protected KeyedCollection(IEqualityComparer<TKey> comparer, int dictionaryCreati
             _threshold = dictionaryCreationThreshold;
         }
 
+        /// <summary>
+        /// Enables the use of foreach internally without allocations using <see cref="List{T}"/>'s struct enumerator.
+        /// </summary>
+        new private List<TItem> Items
+        {
+            get
+            {
+                Debug.Assert(base.Items.GetType() == typeof(List<TItem>));
```

Comment:
Not a big deal since it's an assertion, but `is` is perhaps a closer match to the required invariant, as a subclass wouldn't be incorrect for the rest of the logic.